bluemira/geometry/tools.py

```diff
@@ -758,6 +758,54 @@ def sweep_shape(profiles, path, solid=True, frenet=True, label=""):
     return convert(result, label=label)
 
 
+def fillet_wire_2D(wire: BluemiraWire, radius: float) -> BluemiraWire:
+    """
+    Fillet all edges of a wire
+
+    Parameters
+    ----------
+    wire:
+        Wire to fillet
+    radius:
+        Radius of the fillet operation
+
+    Returns
+    -------
+    filleted_wire:
+        The filleted wire
+    """
+    if radius == 0:
+        return wire.deepcopy()
```

Comment:
I would have raised an error here since the operation cannot be performed. However, if there are particular applications in which this behaviour can be useful, I have nothing in particular against this solution.

Reply:
Yeah that's fair. It seems we've tried to be "forgiving" in most places where we can be. So an offset of 0 also gives a deepcopy. FreeCAD also gives you the same object in this case (but copied), so I just went with the flow